### PR TITLE
Added games/gnoll dice roller

### PR DESCRIPTION
**Pull request by margaretjoanmiller**
## Description of changes

Added the GNOLL dice roller application https://www.ianhunter.ie/GNOLL/.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/gnoll/default.nix
Comment (iblech) [inline pkgs/games/gnoll/default.nix:16]: (I think it's slightly more common to have the two shell commands on individual lines, but I have also seen your stylistic preference. This is just for your information)

File: pkgs/games/gnoll/default.nix
Comment (iblech) [inline pkgs/games/gnoll/default.nix:20]: Could you please remove the trailing space at the end, and shorten to `Efficient dice ...`? We usually don't repeat the name of the package in its description.

File: pkgs/games/gnoll/default.nix
Comment (iblech) [inline pkgs/games/gnoll/default.nix:21]: In the upstream repository it says just GPL 3.0, so `licenses.gpl3Only` would be more accurate. (Or do you have other information?)

File: pkgs/top-level/all-packages.nix
Comment (iblech) [inline pkgs/top-level/all-packages.nix:42350]: There is a separate category for games, starting on line 37511, where according to the top-of-file comment the new entry is supported to go in sorted. (But it's not consistent, so it's easy to get the impression that it's just unsorted.)

File: pkgs/games/gnoll/default.nix
Comment (iblech) [inline pkgs/games/gnoll/default.nix:14]: I think that it's slightly more appropriate to have these as `nativeBuildInputs`, as they are required for compilation and not at runtime (right? I tested it and gnoll seems to work, but I'm not familiar with it).

**Review by iblech — COMMENTED** [review]
Awesome that you packaged gnoll! This looks great! Welcome to the nixpkgs community :wave:

File: pkgs/games/gnoll/default.nix
Comment (iblech) [inline pkgs/games/gnoll/default.nix:22]: Don't forget to add yourself, in the same pull request, to `maintainers/maintainer-list.nix` :-)

**Review by iblech — COMMENTED** [review]
One more note, the usual convention is to use the following commit messages:

maintainers: add 0x4d6165
gnoll: init at 4.3.4

**Comment by 0x4d6165** [comment]
Whoops! didn't mean to close it!

**Comment by iblech** [comment]
Awesome, @0x4d6165, thank you for your continued effort!

I would merge this if I had the permissions, subject to a slight rewrite of history so that there are two commits in total (as in my previous comment). You can carry out this rewrite on your own by using `git rebase -i xxxxxx` where `xxxxxx` is the hash of the newest commit on your branch not authored by you. Then force-push and somebody can hopefully merge this :-)

**Comment by 0x4d6165** [comment]
@iblech done and done!

**Comment by 0x4d6165** [comment]
Sorry, had to do a merge and wasn't sure if I should rebase again

**Comment by iblech** [comment]
Yes, please rebase again: In total, this pull request should comprise exactly two commits (one for the maintainers addition and one for the package). The reason is that the details of the merge process should not clutter the nixpkgs history.

I don't have merge permissions, if nobody else merges this PR you can mention this PR over at https://discourse.nixos.org/t/prs-ready-for-review/3032, I have had good experiences with this procedure in the past. :-) (Of course, as a volunteer-run project, sometimes people don't have available tuits to merge PRs, even if they are already reviewed.)

**Comment by 0x4d6165** [comment]
I don't think I'm doing the rebase properly. That looks like way too many commits and also every time I try to push it makes me merge which adds another commit

**Comment by iblech** [comment]
I agree, 683 commits seem a bit much :-)

Probably the easiest way to fix it is as follows (you can also give me write permissions for your fork and I'll do it), even if it's a bit heavy-handed:

1. You copy the three files you changed to a safe location.
2. `git checkout origin/master`
3. `get pull --ff-only`
4. You add yourself to the maintainers list and run `git commit` with commit message "maintainers: add 0x4d6165"
5. You restore pkgs/games/gnoll/default.nix from the safe location and, manually, add gnoll to pkgs/top-level/all-packages.nix at the correction location, and run `git commit` with message "gnoll: init at 4.3.4"
6. `git checkout -B master`
7. `git push own --force`

This assumes that there are two remotes in .git/config, one named `origin` for NixOS/nixpkgs and one named `own` for your fork. My .git/config looks like this:

```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "origin"]
        url = git@github.com:NixOS/nixpkgs.git
        fetch = +refs/heads/master:refs/remotes/origin/master
[branch "master"]
        remote = origin
        merge = refs/heads/master
[remote "own"]
        url = git@github.com:iblech/nixpkgs.git
        fetch = +refs/heads/*:refs/remotes/own/*
```

File: pkgs/games/gnoll/default.nix
Comment (AndersonTorres) [reply pkgs/games/gnoll/default.nix:16]: This is wrong in all levels.
Just for start, it does not call the pre and post hooks.
```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm755 build/dice $out/bin/
    runHook postInstall
  '';
```

File: pkgs/games/gnoll/default.nix
Comment (AndersonTorres) [inline pkgs/games/gnoll/default.nix:20]:
```suggestion
    description = "An efficient dice notation parser for multiple programming languages that supports a wide set of dice notation";
```

File: pkgs/games/gnoll/default.nix
Comment (AndersonTorres) [inline pkgs/games/gnoll/default.nix:23]: Set platforms

File: pkgs/games/gnoll/default.nix
Comment (AndersonTorres) [inline pkgs/games/gnoll/default.nix:14]: Isn't this a Python package/library? Why not use buildPythonPackage?

File: pkgs/top-level/all-packages.nix
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:37877]:
```suggestion
```

File: pkgs/top-level/all-packages.nix
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:42112]: Use the pkgs/by-name hirerarchy.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
*Restart.*

**Pull request closed without merging**
